app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/LocationConflictModal.tsx

```diff
@@ -0,0 +1,165 @@
+  onCloseClick: () => void
+  cutout: string
+  requiredFixture?: FixtureLoadName
+  requiredModule?: ModuleModel
+}
+
+export const LocationConflictModal = (
+  props: LocationConflictModalProps
+): JSX.Element => {
+  const { onCloseClick, cutout, requiredFixture, requiredModule } = props
+  const { t, i18n } = useTranslation(['protocol_setup', 'shared'])
+  const deckConfig = useDeckConfigurationQuery().data ?? []
+  const { updateDeckConfiguration } = useUpdateDeckConfigurationMutation()
+  const deckConfigurationAtLocationLoadName = deckConfig.find(
+    (deckFixture: Fixture) => deckFixture.fixtureLocation === cutout
+  )?.loadName
+  const currentFixtureDisplayName =
+    deckConfigurationAtLocationLoadName != null
+      ? getFixtureDisplayName(deckConfigurationAtLocationLoadName)
+      : ''
+
+  const handleUpdateDeck = (): void => {
+    if (requiredFixture != null) {
+      updateDeckConfiguration({
+        fixtureLocation: cutout,
+        loadName: requiredFixture,
+      })
+    } else {
+      updateDeckConfiguration({
+        fixtureLocation: cutout,
+        loadName: STANDARD_SLOT_LOAD_NAME,
+      })
+    }
+    onCloseClick()
+  }
+
+  return (
+    <Portal level="top">
+      <LegacyModal
+        title={
+          <Flex
+            flexDirection={DIRECTION_ROW}
+            gridGap={SPACING.spacing10}
+            alignItems={ALIGN_CENTER}
+          >
+            <Icon
+              name="ot-alert"
+              height="1rem"
+              width="1rem"
+              color={COLORS.warningEnabled}
+            />
+            <StyledText as="h3" fontWeight={TYPOGRAPHY.fontWeightSemiBold}>
+              {t('deck_conflict')}
+            </StyledText>
+          </Flex>
+        }
+        onClose={onCloseClick}
+        width="27.75rem"
+      >
+        <Flex flexDirection={DIRECTION_COLUMN}>
+          <Trans
+            t={t}
+            i18nKey="deck_conflict_info"
+            values={{
+              currentFixture: currentFixtureDisplayName,
+              cutout,
+            }}
+            components={{
+              block: <StyledText fontSize={TYPOGRAPHY.fontSizeH4} />,
+              strong: <strong />,
+            }}
+          />
+          <Flex paddingY={SPACING.spacing16} flexDirection={DIRECTION_COLUMN}>
+            <StyledText
+              fontSize={TYPOGRAPHY.fontSizeH4}
+              fontWeight={TYPOGRAPHY.fontWeightBold}
+            >
+              {t('slot_location', { slotName: cutout })}
+            </StyledText>
+            <Flex
+              flexDirection={DIRECTION_COLUMN}
+              paddingTop={SPACING.spacing8}
+              gridGap={SPACING.spacing8}
+            >
+              <Flex
+                padding={SPACING.spacing8}
+                backgroundColor={COLORS.fundamentalsBackground}
+                flexDirection={DIRECTION_ROW}
+                gridGap={SPACING.spacing20}
+              >
+                <StyledText as="p" width="7.8125rem">
```

Comment:
Probably using Flex/Box for width would be better.
```suggestion
                <StyledText as="p" width="7.8125rem">
```

Reply:
i should use flex/box instead of styled text?

Reply:
I mean wrapping `StyledText` with Flex or Box since adding width to p tag isn't recommended.

Reply:
oh i see, yes i can do that!